src/provider/provider_cuda.c

```diff
@@ -294,6 +324,16 @@ static umf_result_t cu_memory_provider_initialize(void *params,
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
+    // If the memory type is shared (CUDA managed), the allocation flags must
+    // be set.
+    // NOTE: we do not check here if the flags are valid - this will be done in
+    // CUDA.
+    if (cu_params->memory_type == UMF_MEMORY_TYPE_SHARED &&
+        cu_params->alloc_flags == 0) {
+        // the default setting is CU_MEM_ATTACH_GLOBAL
+        cu_params->alloc_flags = CU_MEM_ATTACH_GLOBAL;
```

Comment:
Hm, I think it would be better not to modify the params. In theory, following scenario is valid, right?

```
auto params = umfCUDAMemoryProviderParamsCreate();
params.setMemType(UMF_MEMORY_TYPE_SHARED);

provider1 = umfMemoryProviderCreate(..., params);

params.setMemType(UMF_MEMORY_TYPE_DEVICE);
provider2 = umfMemoryProviderCreate(..., params);
```

In this case, the alloc_flags passed for the second provider will be different than what user would expect.

Reply:
Good catch. We should do this for the `cu_provider->alloc_flags`.

Reply:
fixed - I modified only the provider copy of the parameters, not the user. BTW we should change the umfMemoryProviderCreate and make params a const void*